### PR TITLE
WIP initial dump of the experiment

### LINKED ISSUE #28: Update API calls from V1 endpoints
Either move to the latest endpoint, or since we are already bringing in the SDK into the block, change to using that instead

Opinion @Bucknalla ?

### PR DESCRIPTION
**Pull request by curcuz**
WIP!

current experiment aiming at simplifying the code and making it more maintainable. Side goal is to ease the way for a future TS transition.

This PR is a WIP and currently lacks both feature parity ~~and hw revision handling~~. Please do not merge for any reason!

This PR will resolve #28 on merge.

missing features for parity:

* ~~fw download cannot currently handle "latest" and needs instead a specific version passed~~
* ~~balenaFin v1.0~~
* ~~tags~~
* config vars

### CODE REVIEW

**Comment by curcuz** [comment]
@phil-d-wilson @Bucknalla I'm missing the option to create a draft PR - do we need to enable draft PRs in the repo settings?

**Comment by Bucknalla** [comment]
- [ ] Need to change naming from finabler -> fin block

**Comment by curcuz** [comment]
@Bucknalla ready for review - config vars rework is still WIP but I'll address it in a separate PR

File: src/start.sh [hunk 1]
```diff
@@ -1,43 +1,9 @@
 #!/usr/bin/env bash
 
-export DEFAULT_VERSION=2.0.1
-FIRMWARE_PATH=/data/firmware/firmata-
-RELEASE_PATH=https://github.com/balena-io/balena-fin-coprocessor-firmata/releases/download/
-
 if ! test -f /data/firmware/bootloader.s37 ;
```
Comment (Bucknalla) [inline src/start.sh:3]: Does this need to happen in the bash script? I mean its probably fine for now but we could move this to node to be cleaner?

File: src/utils/eeprom/index.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+            const serial = await this.readSerial();
+            const data = await this.parseSerial(serial);
+            debug(data);
+            return data;
+        } catch (error) {
+            throw error;
+        }
+    }
+
+    async writeSerial(serial) {
```
Comment (Bucknalla) [inline src/utils/eeprom/index.js:21]: I wonder if we should protect this behind a 'programming mode' flag. For example, you have to set a config var in order to be able to access this end point?

**Review by Bucknalla — COMMENTED** [review]
Other than these things I think this looks great; @curcuz you can confirm this is working on v1.1; I want to add comphrensive testing in the future but for now this unblocks the testing team and customers so I think after addressing the EEPROM writing concern, this is good to merge!

File: src/start.sh (hunk 1 above)
Comment (Bucknalla) [reply src/start.sh:3]: I meant the bootloader stuff?

Comment (Bucknalla) [reply src/start.sh:3]: @curcuz

Comment (curcuz) [reply src/start.sh:3]: aha! yes, I agree it does make sense to be moved into the js logic - mind filing an issue? I don't feel it should be blocking this PR

**Comment by Bucknalla** [comment]
Looks good to me; ready to approve when you are. Or possibly when balenaCI is happy?

**Review by Bucknalla — APPROVED** [review]
LGTM

**Comment by curcuz** [comment]
@balena-ci retest

**Comment by curcuz** [comment]
re-organized a new PR: #31

**Comment by Bucknalla** [comment]
This was merged with #31 closing as it is now redundant

**Pull request closed without merging**